h/api/schemas.py

```diff
@@ -264,17 +192,50 @@ class CreateAnnotationSchema(object):
     """Validate the POSTed data of a create annotation request."""
 
     def __init__(self, request):
-        self.structure = AnnotationSchema(request)
+        self.structure = AnnotationSchema()
+        self.request = request
 
     def validate(self, data):
-        return self.structure.validate(data)
+        appstruct = self.structure.validate(data)
+
+        new_appstruct = {}
+
+        _remove_protected_fields(appstruct)
+
+        new_appstruct['userid'] = self.request.authenticated_userid
+        new_appstruct['target_uri'] = appstruct.pop('uri', u'')
+        new_appstruct['text'] = appstruct.pop('text', u'')
```

Comment:
Yes. Was just going through the default values we set here and the default values/(not-)NULL constraints on the database. We do allow `NULL` on the `text` column, so there's no need to store an empty string. At some later point we might want to assume that an empty string is considered `NULL` as well, which would make querying easier. But that might have other implications and is a whole different discussion :)